src/main/java/org/acme/order/service/OrderService.java

```diff
@@ -45,38 +45,36 @@ public class OrderService {
     * @param info The order information.
     * @return A created Order with incoming info, new unique identifier and created status.
     * @throws UnavailablePastryException
-    * @throws Exception
     */
-   public Order placeOrder(OrderInfo info) throws UnavailablePastryException, Exception {
+   public Order placeOrder(OrderInfo info) throws UnavailablePastryException {
       // For all products in order, check the availability calling the Pastry API.
       var availabilityFutures = info.productQuantities().stream()
           .map(ProductQuantity::productName)
-          .collect(Collectors.toMap(this::checkPastryAvailability, Function.identity()));
+          .map(this::checkPastryAvailability)
+          .toList();
 
-      // Wait for all completable future to finish.
-      CompletableFuture.allOf(availabilityFutures.keySet().toArray(new CompletableFuture[0])).join();
+      // Wait for all completable futures to finish.
+      CompletableFuture.allOf(availabilityFutures.toArray(new CompletableFuture[availabilityFutures.size()])).join();
 
-      try {
-         // If one pastry is marked as unavailable, throw a business exception.
-         for (CompletableFuture<Boolean> availabilityFuture : availabilityFutures.keySet()) {
-            if (!availabilityFuture.get()) {
-               String pastryName = availabilityFutures.get(availabilityFuture);
-               throw new UnavailablePastryException(pastryName, "Pastry " + pastryName + " is not available");
-            }
-         }
-      } catch (InterruptedException | ExecutionException e) {
-         throw new Exception("Unexpected exception: " + e.getMessage());
+      var unavailablePastry = availabilityFutures.stream()
+          .map(this::getAvailability)
+          .filter(availability -> !availability.available())
+          .findAny();
+
+      if (unavailablePastry.isPresent()) {
+         var pastryName = unavailablePastry.get().pastryName();
+         throw new UnavailablePastryException(pastryName, "Pastry %s is not available".formatted(pastryName));
       }
 
       // Everything is available! Create (and probably persist ;-) a new order.
-      Order result = new Order();
+      var result = new Order();
```

Comment:
Just for personal understanding: why switch to a `var`? Is there some technical optimization reason? Or personal preferences? I didn't make the switch yet, still sticking with types... I'd like to understand 😉

Reply:
I think thats the OCD in me. I stopped doing types back in Java 11 :) I can revert if you'd prefer. Using types isn't wrong.

Reply:
No no that's good! OCD is fine 😉